src/Catalyst.KBucket/Bucket.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Catalyst.KBucket
+{
+    /// <summary>
+    ///   A binary tree node in the <see cref="Catalyst.KBucket{T}"/>.
+    /// </summary>
+    public class Bucket<T>
+        where T : class, IContact
+    {
+        /// <summary>
+        ///   The contacts in the bucket.
+        /// </summary>
+        public List<T> Contacts = new List<T>();
+
+        /// <summary>
+        ///  Determines if the bucket can be split.
+        /// </summary>
+        public bool DontSplit;
+
+        /// <summary>
+        ///   The left hand node.
+        /// </summary>
+        public Bucket<T> Left;
+
+        /// <summary>
+        ///   The right hand node.
+        /// </summary>
+        public Bucket<T> Right;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)